docs/governance/rfc/004-dart-implementation.md

```diff
@@ -0,0 +1,63 @@
+# 004: Dart Implementation
+
+This RFC proposes adding a Dart implementation of Connect starting with support for clients.
+This will be similar in scope to Connect-Swift and Connect-Kotlin implementations.
+
+## Why
+
+Dart is a popular programming language powering frameworks like Flutter which
+is used for cross-platform app development. This will help grow Connect's
+adoption.
+
+## Anticipated complications
+
+Dart runs on several platforms: Web, Android, and iOS to name a few. Most of the platforms
+have their own HTTP stack with varying levels of support for HTTP specs. It doesn't have a
+unified abstration and APIs for common HTTP constructs like headers and cancellation.
+
+We have to create a minimal HTTP abstraction to support various platforms. We can then have concrete
+implementations using the following:
+
+- [`HttpClient`][dart-io-client] from `dart:io`. This is part of the std library and runs on native platforms.
+  This only supports HTTP/1.
+- [`package:http2`][pub-http2]. This is a HTTP/2 implementation maintained by the Dart team. This can be used
+  to support gRPC. Connection management has to be implemented on top of this.
+- [`dart:js_interop`][dart-js-interop]: This is part of the std library and can be used to create
+  bindings for web APIs. We can use the `fetch` APIs to support Connect and gRPC-Web.
+
+Platforms like Android and iOS have their own HTTP stack, and it is desirable to use them. [`package:http`][pub-http] 
+can be used as a cross platform client to support both. It is maintained by the Dart team. The only blocker
+is the missing support for [aborting requests][http-abort-issue].
```

Comment:
Will this be an actual _blocker_? If so, what will use instead for an initial release? Or will we use this but it blocks promoting to v1.0 until we can support aborting requests? Maybe clarify a little how this issue impacts how we intend to implement support for Android and iOS.